### PR TITLE
Backport: VNTags decoder (disabled by default)

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #6117 

Backport VNTag decoder to master-6.0.x

Describe changes:
- clang-format fixups
- [4489](https://redmine.openinfosecfoundation.org/issues/4489)

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: src/decode-vntag.c [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+ * \param tv pointer to the thread vars
+ * \param dtv pointer to decode thread vars
+ * \param p pointer to the packet struct
+ * \param pkt pointer to the raw packet
+ * \param len packet len
+ *
+ */
+int DecodeVNTag(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p, const uint8_t *pkt, uint32_t len)
+{
+    if (unlikely(!g_vntag_enabled))
```
Comment (victorjulien) [inline src/decode-vntag.c:70]: its unlikely that we don't have it enabled, but its disabled by default?

File: rules/decoder-events.rules [hunk 2]
```diff
@@ -113,7 +115,6 @@ alert icmp any any -> any any (msg:"SURICATA ICMPv4 invalid checksum"; icmpv4-cs
 alert tcp any any -> any any (msg:"SURICATA TCPv6 invalid checksum"; tcpv6-csum:invalid; classtype:protocol-command-decode; sid:2200077; rev:2;)
 alert udp any any -> any any (msg:"SURICATA UDPv6 invalid checksum"; udpv6-csum:invalid; classtype:protocol-command-decode; sid:2200078; rev:2;)
 alert icmp any any -> any any (msg:"SURICATA ICMPv6 invalid checksum"; icmpv6-csum:invalid; classtype:protocol-command-decode; sid:2200079; rev:2;)
-
```
Comment (victorjulien) [inline rules/decoder-events.rules:116]: unrelated whitespace change

File: src/decode-vntag.c [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        ENGINE_SET_INVALID_EVENT(p, VNTAG_HEADER_TOO_SMALL);
+        return TM_ECODE_FAILED;
+    }
+
+    if (!PacketIncreaseCheckLayers(p)) {
+        return TM_ECODE_FAILED;
+    }
+
+    VNTagHdr *vntag_hdr = (VNTagHdr *)pkt;
+    if (vntag_hdr == NULL)
```
Comment (victorjulien) [inline src/decode-vntag.c:85]: Created https://redmine.openinfosecfoundation.org/issues/4496 to figure out a better way to deal with this pattern. I think `pkt` can never be `NULL`, but lets first change this in `master` and see if it can indeed be removed.

**Review by victorjulien — CHANGES_REQUESTED** [review]
I'd like to see 2 Suricata-Verify tests: one that validates that the decoder works and one that validates that disabling it (or leaving it disabled actually) works.

**Comment by jlucovsky** [comment]
> I'd like to see 2 Suricata-Verify tests: one that validates that the decoder works and one that validates that disabling it (or leaving it disabled actually) works.
Done -- see s-v pr #500

Continued in #6125
